packages/react-ui-map/src/Mapbox/Mapbox.js

```diff
@@ -41,20 +40,16 @@ export default class Mapbox extends PureComponent {
   getChildContext() {
     return {
       map: this.state.map,
+      MapboxGL: this.MapboxGL,
     }
   }
 
-  componentDidMount() {
-    const opts = {
-      container: this.container,
-      style: this.props.style,
-      zoom: this.props.zoom,
-      theme: this.props.theme,
-    }
+  async componentDidMount() {
+    /* Load mapbox-gl on client only.
+       Attempting to load mapbox-gl on server will break server-side rendering. */
+    this.MapboxGL = await import('mapbox-gl/dist/mapbox-gl')
```

Comment:
You might also need a wrapper around this.  Say this component unmounted.  I might assume this would load in mapbox again and definitely run `this.setupMapbox()` again.  Something like this?
```
if (window && !window.mapboxgl) {
  // import
}
```

You could also use this anywhere else in code.  It will still work for both SSR and client.

Not sure why it's not available on rent-js right now, but through the [examples](https://www.mapbox.com/mapbox-gl-js/examples/) it is and it's something I've used before.

![screen shot 2017-09-14 at 7 29 19 am](https://user-images.githubusercontent.com/10748767/30435431-9965f7ea-991e-11e7-96b5-70a6c250f1ee.png)

Also, there's also this one which we _really_ should be using as well.  I think we used it on rent in flightjs

https://github.com/mapbox/mapbox-gl-supported

Reply:
I'll have to check. I would assume that `import()` would know whether something is loaded/cached, or at the very least the browser would still have the JS cached. It should be easy to check though, just switch from Map view to List view and back. But I think we need to call `setupMapbox()` when remounted anyway, to get an instance of the Map

As for the `mapbox-gl-supported`, not really sure what we would do there. We have nothing to fall back to so the user is out of luck I think.

Reply:
yep, agreed on the call of `setupMapbox()`.